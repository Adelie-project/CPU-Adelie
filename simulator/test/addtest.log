read 56 bytes.


PC = 00000000 
r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:00000000 r07:00000000 
r08:00000000 r09:00000000 r10:00000000 r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000000 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 
press 's' to step-in, others to run. addi r6 r0 $1000


PC = 00000004 
r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:000003E8 r07:00000000 
r08:00000000 r09:00000000 r10:00000000 r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000000 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 
press 's' to step-in, others to run. addi r7 r0 $2000


PC = 00000008 
r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:000003E8 r07:000007D0 
r08:00000000 r09:00000000 r10:00000000 r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000000 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 
press 's' to step-in, others to run. add r8 r6 r7


PC = 0000000C 
r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:000003E8 r07:000007D0 
r08:00000BB8 r09:00000000 r10:00000000 r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000000 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 
press 's' to step-in, others to run. add r6 r7 r8


PC = 00000010 
r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:00001388 r07:000007D0 
r08:00000BB8 r09:00000000 r10:00000000 r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000000 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 
press 's' to step-in, others to run. add r7 r8 r6


PC = 00000014 
r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:00001388 r07:00001F40 
r08:00000BB8 r09:00000000 r10:00000000 r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000000 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 
press 's' to step-in, others to run. add r8 r6 r7


PC = 00000018 
r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:00001388 r07:00001F40 
r08:000032C8 r09:00000000 r10:00000000 r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000000 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 
press 's' to step-in, others to run. add r6 r7 r8


PC = 0000001C 
r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:00005208 r07:00001F40 
r08:000032C8 r09:00000000 r10:00000000 r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000000 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 
press 's' to step-in, others to run. add r7 r8 r6


PC = 00000020 
r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:00005208 r07:000084D0 
r08:000032C8 r09:00000000 r10:00000000 r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000000 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 
press 's' to step-in, others to run. lui r10 $2147418112


PC = 00000024 
r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:00005208 r07:000084D0 
r08:000032C8 r09:00000000 r10:7FFF0000 r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000000 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 
press 's' to step-in, others to run. add r10 r10 r7


PC = 00000028 
r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:00005208 r07:000084D0 
r08:000032C8 r09:00000000 r10:7FFF84D0 r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000000 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 
press 's' to step-in, others to run. addi r10 r10 $-465


PC = 0000002C 
r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:00005208 r07:000084D0 
r08:000032C8 r09:00000000 r10:7FFF82FF r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000000 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 
press 's' to step-in, others to run. addi r10 r10 $-2000


PC = 00000030 
r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:00005208 r07:000084D0 
r08:000032C8 r09:00000000 r10:7FFF7B2F r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000000 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 
press 's' to step-in, others to run. add r10 r7 r10


PC = 00000034 
r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:00005208 r07:000084D0 
r08:000032C8 r09:00000000 r10:7FFFFFFF r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000000 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 
press 's' to step-in, others to run. jal r20 $0

PC = 00000034
r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:00005208 r07:000084D0 
r08:000032C8 r09:00000000 r10:7FFFFFFF r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000038 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 

program infinitely loops at pc 52, simulation stops
