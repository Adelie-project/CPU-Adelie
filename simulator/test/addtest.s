  addi r6, r0, $1000 ; int a = 1000
  addi r7, r0, $2000 ; int b = 2000
  add r8, r6, r7    ; int c = a + b = 3000
  add r6, r7, r8    ; int a = b + c = 5000
  add r7, r8, r6    ; int b = c + a = 8000
  add r8, r6, r7    ; int c = a + b = 13000
  add r6, r7, r8    ; int a = b + c = 21000
  add r7, r8, r6    ; int b = c + a = 34000
  lui r10, $0x7FFF0000 ; int x = 0x7FFF0000
  add r10, r10, r7  ; int x = 0x7FFF84D0
  addi r10, r10, $-465
  addi r10, r10, $-2000
  add r10, r7, r10  ; int x = 0x7FFFFFFF
  jal r20, $0     ; stop at here
