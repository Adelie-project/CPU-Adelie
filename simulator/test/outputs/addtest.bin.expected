Now additional value of PC is '4'.
If this is as intended, enter 'r', others to terminate.


PC = 00000034

r00:00000000 r01:00000000 r02:00000000 r03:00000000 r04:00000000 r05:00000000 r06:00005208 r07:000084D0 
r08:000032C8 r09:00000000 r10:7FFFFFFF r11:00000000 r12:00000000 r13:00000000 r14:00000000 r15:00000000 
r16:00000000 r17:00000000 r18:00000000 r19:00000000 r20:00000038 r21:00000000 r22:00000000 r23:00000000 
r24:00000000 r25:00000000 r26:00000000 r27:00000000 r28:00000000 r29:00000000 r30:00000000 r31:00000000 

program infinitely loops at pc 52, simulation stops
